core/src/display_object/graphic.rs

```diff
@@ -93,6 +93,12 @@ impl<'gc> Graphic<'gc> {
             },
         ))
     }
+
+    pub fn drawing(&self, gc_context: MutationContext<'gc, '_>) -> RefMut<'_, Drawing> {
+        RefMut::map(self.0.write(gc_context), |w| {
+            w.drawing.get_or_insert_with(Drawing::new)
```

Comment:
side observation: I feel like we should make Drawing either always-eagerly-constructed (I think there is only one place where we _don't_ construct it eagerly currently), or always-lazy (and, in fact, hidden behind an `Option<Box>`), not a mix. Given how relatively rare drawing ops are and Drawing takes 200+B per DO, the latter sounds

Reply:
Agree, seems like currently `Drawing` is eagerly-initialized only in `Graphic::new_with_avm2()`, which is called from AVM2's `flash.display.Shape` constructor. Do you think changing to `Option<Box<Drawing>>` should be part of this PR?

Reply:
> Agree, seems like currently Drawing is eagerly-initialized only in Graphic::new_with_avm2(), which is called from AVM2's flash.display.Shape constructor. Do you think changing to Option<Box<Drawing>> should be part of this PR?

Up to you. I feel like it'd touch all the same spots in code, so might as well do it here?

Reply:
I won't have much time this week, so I prefer doing this in a separate PR.